woocommerce/woocommerce
# Woocommerce dashboard status widget net sales link and logic

### All Submissions:


### Changes proposed in this Pull Request:

Updated the net sales link when Woocommerce Admin is not disabled to direct to the new analytics screen.
Also added a `WC_Admin_Report_V2` that overwrites the `sales_sparkline` method to make use of data coming from the new WooAdmin api with `rest_do_request`.
Also added a `get_performance_data` function that returns the necessary sales report data in a similar form as what `get_sales_report_data` currently does, so that they can be seemingly interchangeable for now.

I am open for other idea's on the above class, as it could easily be stand alone vs an extended class.

Closes #28569 

[Gif when Woo Admin is disabled](https://d.pr/i/R2Yvjg)

[Gif when Woo Admin is enabled](https://d.pr/i/hiwURU)

### How to test the changes in this Pull Request:

1.  Pull down this branch and make sure Woo is enabled
2. Navigate to the WP dashboard (**Dashboard > Home**) and scroll down to the **WooCommerce Status** widget
3. Click on the first row (net sales this month), it should redirect to `wp-admin/admin.php?page=wc-admin&path=%2Fanalytics%2Frevenue&chart=net_revenue&orderby=net_revenue&period=month&compare=previous_period`
4. Now add a product and a completed order with the product
5. Make sure all the actions are completed in **WooCommerce > Status > Scheduled Actions**
6. Go back to the dashboard WooCommerce status widget. The net sales should now show an updated number, and the little spark line on the right should match accordingly (have at-least one spike of the completed order).
7. Click on the link again, and see if the displayed amount matches that of the **Analytics > Revenue** screen.
8. Go back to the dashboard a second line should be present with the top performing product
9. Click on the top seller product name, it should redirect to `wp-admin/admin.php?page=wc-admin&filter=single_product&products=PRODUCT_ID&path=%2Fanalytics%2Fproducts` where the product is selected under the `show` dropdown.
10. Add `add_filter( 'woocommerce_admin_disabled', '__return_true' );` to your themes `functions.php` file.
11. Load the WP dashboard again
12. The above buttons should now be directing to the old **WooCommerce > Reports** screen, it should still show the correct number in the widget.

### Other information:



### Changelog entry

> Enhancement - Update the Woo widget net sales link and logic to use the new analytics page and data.
